docs/LabVIEW_Debugging_Techniques.md

```diff
@@ -0,0 +1,128 @@
+### Debugging and Benchmarking
+
+Debugging and benchmarking is a normal process of code development. There are a variety of ways to debug and benchmark custom devices.
+
+#### LabVIEW Debugging Techniques
+
+Custom devices are written in LabVIEW code. Therefore it’s possible to develop, test and debug in LabVIEW before running the Custom Device Template Tool. In other words, you can use **[LabVIEW’s built-in debugging techniques](https://www.ni.com/en-us/support/documentation/supplemental/12/debugging-techniques-in-labview.html)** during development; and merge the LabVIEW code into the custom device framework after it matures.
+
+Since the custom device is one of many parts of the system definition, the behavior of the LabVIEW code within the custom device framework will likely differ from the stand-alone LabVIEW application, especially in regards to timing. As a result, you should benchmark the custom device inside of the VeriStand Engine.
+Once added to the system definition, custom devices have been fully integrated into VeriStand’s context. As a result, LabVIEW’s built-in debugging techniques are no longer available. Several techniques are available for debugging and benchmarking the custom device.
+ 
+#### Console Viewer
+
+A subcomponent of VeriStand RT Engine is the RT Console Viewer.
+You can install it to the execution host using Measurement and Automation Explorer. When installed, the component runs a small UDP daemon allowing the operator to view the console from a utility called the RT Console Viewer. You can access the RT Console viewer from **VeriStand » Workspace » Tools » Console Viewer**.	
+
+![](images/ConsoleViewer1.jpg)
+
+**Figure: RT Console Viewer**
+
+The Console Viewer will show the system definition and the resulting CPU usage. The viewer is also useful for displaying debugging messages. The console viewer provides a periodic snapshot of utilization. CPU spikes and transients will probably be unobservable. If the system is very busy, it may not update the console viewer at all. You can use other debugging methods for a more accurate indication of resource utilization.
+
+As the name implies, the RT Console viewer is only available on real-time targets. The RT Console Viewer is also available as a stand-alone add-on to LabVIEW Real-Time. See **[NI Developer Zone Tutorial: Remotely View Console Output of Real-Time Targets](https://knowledge.ni.com/KnowledgeArticleDetails?id=kA03q000000x4TjCAI&l=en-US)** for more information.
+
+#### Printing to the Console
+Printing to the console is often all that’s needed to debug an application.
+
+#### Printing With NIVS Debug String VI
+The recommended method of printing to the console is to use Print NIVS Debug String VI. You can download the VI from NI Community » VeriStand Add-Ons » Documents » Print VeriStand Debug String.
+
+![](images/NIVSdebugString.jpg)
```

Comment:
Please replace the photo of the VI with this one:
![image](https://user-images.githubusercontent.com/62251693/132237918-3831c5bd-fe7e-4d5d-aa83-cf3d36b064c6.png)